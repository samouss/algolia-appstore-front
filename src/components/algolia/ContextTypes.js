import PropTypes from 'prop-types';

export default {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};
