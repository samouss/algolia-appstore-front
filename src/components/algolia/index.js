export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
