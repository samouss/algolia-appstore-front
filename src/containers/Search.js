import React from 'react';
import PropTypes from 'prop-types';
import { withQuery } from 'components/Algolia';
import { TextInput } from 'components/Input';

export const Search = ({ query, onChange, ...props }) => (
  <TextInput
    {...props}
    value={query}
    onChange={event => onChange(event.currentTarget.value)}
    placeholder="Angry Birds, Cut the Rope, Pinterest..."
  />
);

Search.propTypes = {
  query: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default withQuery(Search);
