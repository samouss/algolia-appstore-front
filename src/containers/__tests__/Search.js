import React from 'react';
import { shallow } from 'enzyme';
import { createMockAlgoliaClient, createMockAlgoliaHelper } from 'test/algolia';
import Search from '../Search';

describe('<Search />', () => {
  it('expect to render', () => {
    const context = {
      algoliaClient: createMockAlgoliaClient(),
      algoliaHelper: createMockAlgoliaHelper(),
    };

    const component = shallow(
      <Search />,
      { context },
    );

    expect(component).toMatchSnapshot();
  });
});
