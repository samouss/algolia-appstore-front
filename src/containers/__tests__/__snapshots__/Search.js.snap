// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> expect to render 1`] = `
<Search
  onChange={[Function]}
  query=""
/>
`;
